ephios/core/services/notifications/types.py

```diff
@@ -530,29 +530,34 @@ class CustomEventParticipantNotification(AbstractNotificationHandler):
     @classmethod
     def send(cls, event: Event, content: str):
         participants = set()
+        notifications = []
         responsible_users = get_users_with_perms(
             event, with_superusers=False, only_with_perms_in=["change_event"]
         )
         for shift in event.shifts.all():
-            participants.update(shift.get_participants())
-        notifications = []
-        for participant in participants:
-            user = participant.user if isinstance(participant, LocalUserParticipant) else None
-            if user in responsible_users:
-                continue
-            notifications.append(
-                Notification(
-                    slug=cls.slug,
-                    user=user,
-                    data={
-                        "email": participant.email,
-                        "event_id": event.id,
-                        "content": content,
-                        "event_title": event.title,
-                        "event_url": participant.reverse_event_detail(event),
-                    },
-                )
-            )
+            for participation in shift.participations.filter(
+                state__in={AbstractParticipation.States.CONFIRMED}
+            ):
```

Comment:
what about
```python
AbstractParticipation.objects.filter(shift__event=event, state=AbstractParticipation.States.CONFIRMED)
```
to only have one for loop?

Reply:
nice one